classes/Customer.php

```diff
@@ -265,6 +269,33 @@ public function add($autoDate = true, $nullValues = true)
         return $success;
     }
 
+    /**
+     * Adds current Customer as a new Object to the database.
+     *
+     * @param bool $autoDate Automatically set `date_upd` and `date_add` columns
+     * @param bool $nullValues Whether we want to use NULL values instead of empty quotes values
+     *
+     * @return bool Indicates whether the Customer has been successfully added
+     *
+     * @throws PrestaShopDatabaseException
+     * @throws PrestaShopException
+     */
+    public function addWs($autodate = true, $null_values = false)
+    {
+        if (CustomerCore::customerExists($this->email)) {
```

Comment:
You should use `Customer` instead of `CustomerCore` 
```suggestion
        if (Customer::customerExists($this->email)) {
```